crypto/keyring/keyring.go

```diff
@@ -426,6 +432,30 @@ func (ks keystore) DeleteByAddress(address sdk.Address) error {
 	return nil
 }
 
+func (ks keystore) Rename(oldName, newName string) error {
+	_, err := ks.Key(newName)
+	if err == nil {
+		return fmt.Errorf("rename failed: %s already exists in the keyring", newName)
+	}
+
+	armor, err := ks.ExportPrivKeyArmor(oldName, passPhrase)
+	if err != nil {
+		return err
+	}
+
+	err = ks.Delete(oldName)
+	if err != nil {
+		return err
+	}
```

Comment:
Should this be called after `ImportPrivKey`? What happens if `ImportPrivKey` fails? The account is deleted and no new account is created.

Reply:
so import will actually fail if you delete last, cause import doesn't allow same pubkeys. some solutions i can think of:

1) dump the key information in a recovery file

2) update the ImportPrivKey method to allow keys with same pubkeys, but different keyring names

thoughts?

Reply:
If the `name` is the primary index, I think it's OK to go with (2).

Reply:
alright, [changed](https://github.com/cosmos/cosmos-sdk/pull/9601/files#diff-6832e2d710aed661f9b5941a88ac18045bb2555d7d9be1319f403dbe928017b4R819-R835) the write behavior to allow duplicate keys, as long as the names are different.